src/solver/simulation/sim_structure_probleme_economique.h

```diff
@@ -478,11 +489,14 @@ struct PROBLEME_HEBDO
     char OptimisationAvecCoutsDeDemarrage;
     int NombreDePays;
     const char** NomsDesPays;
+    AdequacyPatchMode* AreaAdequacyPatchMode;
     int NombreDePaliersThermiques;
 
     int NombreDInterconnexions;
     int* PaysOrigineDeLInterconnexion;
     int* PaysExtremiteDeLInterconnexion;
+    AdequacyPatchMode* StartAreaAdequacyPatchType;
+    AdequacyPatchMode* EndAreaAdequacyPatchType;
     COUTS_DE_TRANSPORT** CoutDeTransport;
```

Comment:
Also, instead of having two arrays for the link types (`StartAreaAdequacyPatchType` and `EndAreaAdequacyPatchType`), you could have a single array/vector containing objects of this new type
```cpp
struct {
   AdequacyPatchMode origin;
   AdequacyPatchMode extremity;
} AdqPatchLinkType;